src/main/java/executor/service/service/impl/ExecutionServiceImpl.java

```diff
@@ -0,0 +1,37 @@
+package executor.service.service.impl;
+
+import executor.service.model.ProxyConfigHolder;
+import executor.service.model.Scenario;
+import executor.service.model.WebDriverConfig;
+import executor.service.service.ExecutionService;
+import executor.service.service.ScenarioExecutor;
+import org.openqa.selenium.WebDriver;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ExecutionServiceImpl implements ExecutionService {
```

Comment:
Why are you not satisfied with my implementation of this service? See Pull-Request https://github.com/GFLCourses5/worker/pull/46.

You could apply it here.

public class ExecutionServiceImpl implements ExecutionService {

    private static final Logger log = LoggerFactory.getLogger(ExecutionServiceImpl.class);

    private final ScenarioExecutor scenarioExecutor;
    private final WebDriverConfig webDriverConfig;
    private final ProxyConfigHolder defaultProxy;

    public ExecutionServiceImpl(ScenarioExecutor scenarioExecutor,
                                WebDriverConfig webDriverConfig, ProxyConfigHolder defaultProxy) {
        this.scenarioExecutor = scenarioExecutor;
        this.webDriverConfig = webDriverConfig;
        this.defaultProxy = defaultProxy;
    }

    /**
     * Execute ScenarioExecutor.
     *
     * @param scenarios the queue with scenarios
     * @param proxies   the queue with proxies
     */
    @Override
    public void execute(BlockingQueue<Scenario> scenarios,
                        BlockingQueue<ProxyConfigHolder> proxies) {
        ProxyConfigHolder proxy = defaultProxy;
        while (true) {
            try {
                Scenario scenario = scenarios.take();

                ProxyConfigHolder newProxy = proxies.poll();
                if (newProxy != null) {
                    proxy = newProxy;
                }

                WebDriver webDriver = getWebDriverPrototype(webDriverConfig, proxy);
                if (webDriver == null) continue;

                scenarioExecutor.execute(scenario, webDriver);
            } catch (InterruptedException e) {
                log.error("Thread was interrupted in ExecutionServiceImpl.class", e);
                Thread.currentThread().interrupt();
            }
        }
    }

    /**
     * Get WebDriver as Prototype.
     *
     * @param webDriverConfig   the WebDriverConfig entity
     * @param proxyConfigHolder the ProxyConfigHolder entity
     */
    private WebDriver getWebDriverPrototype(WebDriverConfig webDriverConfig, ProxyConfigHolder proxyConfigHolder) {
        WebDriverInitializer webDriverInitializer = new WebDriverInitializerImpl();
        return webDriverInitializer.getInstance(webDriverConfig, proxyConfigHolder);
    }
}